python/GafferUI/PlugValueWidget.py

```diff
@@ -299,14 +299,22 @@ def _popupMenuDefinition( self ) :
 			menuDefinition.append( "/CopyPasteDivider", { "divider" : True } )
 
 		if self.getPlug().getInput() is not None :
-			menuDefinition.append( "/Edit input...", { "command" : Gaffer.WeakMethod( self.__editInput ) } )
-			menuDefinition.append( "/EditInputDivider", { "divider" : True } )
-			menuDefinition.append(
-				"/Remove input", {
-					"command" : Gaffer.WeakMethod( self.__removeInput ),
-					"active" : self.getPlug().acceptsInput( None ) and not self.getReadOnly() and not Gaffer.MetadataAlgo.readOnly( self.getPlug() ),
-				}
-			)
+			if not Gaffer.Animation.isAnimated( self.getPlug() ) :
```

Comment:
I'm wondering if we should deal with this in `AnimationUI.__popupMenu()` rather than in here, to avoid an unnecessary coupling and keep all the animation menu items in one place. I guess the AnimationUI would then have to delete the "Remove input" item, so it's still not perfect, but it's probably better to have the higher level component (AnimationUI) know about the lower level component (PlugValueWidget) rather than vice-versa?